spec/02-integration/07-sdk/06-worker_events_spec.lua

```diff
@@ -0,0 +1,144 @@
+local helpers = require "spec.helpers"
+
+describe("worker_events", function()
+  local strategy = "off"
+  local test_cases = {"string", "table", }
+  local business_port
+
+  lazy_setup(function()
+    business_port = helpers.get_available_port()
+    local fixtures = {
+      http_mock = {
+        worker_events = [[
+          server {
+            server_name example.com;
+            listen %s;
+
+            location = /test_too_big_string {
+              content_by_lua_block {
+                local PAYLOAD_TOO_BIG_ERR = "failed to publish event: payload too big"
+                local SOURCE, EVENT = "foo", "string"
+                local payload_received = ""
+                local worker_events = kong.worker_events
+                local cjson = require "cjson.safe"
+
+                local function generate_data()
+                  return string.rep("X", 70000)
+                end
+
+                local function wait_until(validator, timeout)
+                  local deadline = ngx.now() + (timeout or 5)
+                  local res
+                  repeat
+                    worker_events.poll()
+                    res = validator()
+                  until res or ngx.now() >= deadline
+                  return res
+                end
+
+                -- subscribe
+                local ok, err = worker_events.register(function(data)
+                  payload_received = data
+                end, SOURCE, EVENT)
+
+                -- we look forward to receiving the payload even if 
+                -- the size of the payload exceeds the limit
+
+                -- when payload is a string
+                local PAYLOAD = generate_data()
+                local ok, err = worker_events.post(SOURCE, EVENT, PAYLOAD)
+                if not ok then
+                  ngx.status = ngx.HTTP_INTERNAL_SERVER_ERROR
+                  ngx.say("post string failed, err: " .. cjson.encode(err))
```

Comment:
It seems that we do not check the response body, could we remove these `ngx.say()` calls? Or add `assert.equal(res.body, "ok")`?

Reply:
done